Description: Search for mod files in $FMODDIR directory
 Also look in a fortran-ABI-specific directory for libs
Author: Alastair McKinstry <mckinstry@debian.org>
Last-Updated: 2023-12-21
Forwarded: no


--- a/ompi/tools/wrappers/mpifort-wrapper-data.txt.in
+++ b/ompi/tools/wrappers/mpifort-wrapper-data.txt.in
@@ -13,7 +13,7 @@
 compiler_flags_env=FCFLAGS
 compiler=@FC@
 preprocessor_flags=
-compiler_flags=@OMPI_WRAPPER_EXTRA_FCFLAGS@
+compiler_flags=-I@FMODDIR@/openmpi
 linker_flags=@OMPI_WRAPPER_EXTRA_FC_LDFLAGS@
 # Note that per https://svn.open-mpi.org/trac/ompi/ticket/3422, we
 # intentionally only link in the MPI libraries (ORTE, OPAL, etc. are
@@ -24,5 +24,8 @@
 dyn_lib_file=lib@OMPI_LIBMPI_NAME@.@OPAL_DYN_LIB_SUFFIX@
 static_lib_file=lib@OMPI_LIBMPI_NAME@.a
 required_file=@OMPI_WRAPPER_FORTRAN_REQUIRED_FILE@
+# The same includedir is used in mpif77 and mpif90; so link the mod files in there
+# but set fmoddir and use it where possible
+fmoddir=@FMODDIR@
 includedir=${includedir}
-libdir=${libdir}
+libdir=${libdir}/fortran/@FC@
